### PR TITLE
add auto labeling of pull-requests

### PR DESCRIPTION
**Pull request by maxandersen**
This is a basic first attempt on autolabeler for pullrequests.

I just matched labels against folder names to get an initial setup. Not all matched; but wee can tweak it as we go.

note: I generated this in late december so if new extensions or integraitons tests folders was added then they aren't listed in here.

### CODE REVIEW

File: .github/labeler.yml [hunk 1]
```diff
@@ -0,0 +1,273 @@
...
+
+area/config:
+ - extensions/config-yaml/**/*
+
+area/documentation:
+ - docs/**/*
+
+area/hibernate-orm:
+ - extensions/hibernate-orm/**/*
+ - integration-tests/hibernate-orm-panache/**/*
```
Comment (gsmet) [inline .github/labeler.yml]: You also can add all the `jpa-*` ITs

File: .github/labeler.yml [hunk 2]
```diff
@@ -0,0 +1,273 @@
...
+ - integration-tests/hibernate-search-elasticsearch/**/*
+
+area/hibernate-validator:
+ - extensions/hibernate-validator/**/*
+ - integration-tests/hibernate-validator/**/*
+
+area/infinispan:
+ - extensions/infinispan-client/**/*
+ - extensions/infinispan-embedded/**/*
+ - integration-tests/infinispan-cache-jpa/**/*
```
Comment (gsmet) [inline .github/labeler.yml]: You can add this one to Hibernate too. It needs both labels IMHO.

File: .github/labeler.yml [hunk 3]
```diff
@@ -0,0 +1,273 @@
...
+ - integration-tests/kubernetes-client/**/*
+
+area/logging:
+ - extensions/logging-gelf/**/*
+ - extensions/logging-json/**/*
+ - extensions/logging-sentry/**/*
+ - integration-tests/logging-gelf/**/*
+
+area/maven:
+ - integration-tests/maven/**/*
```
Comment (gsmet) [inline .github/labeler.yml]: Shouldn't we add the Maven plugin directory too?

File: .github/labeler.yml [hunk 4]
```diff
@@ -0,0 +1,273 @@
...
+
+area/mongodb:
+ - extensions/mongodb-client/**/*
+ - integration-tests/mongodb-client/**/*
+ - integration-tests/mongodb-panache/**/*
```
Comment (gsmet) [inline .github/labeler.yml]: The MongoDB Panache extension should be added here too.

File: .github/labeler.yml [hunk 5]
```diff
@@ -0,0 +1,273 @@
...
+area/persistence:
+ - integration-tests/jpa/**/*
+ - integration-tests/jpa-derby/**/*
+ - integration-tests/jpa-h2/**/*
+ - integration-tests/jpa-mariadb/**/*
+ - integration-tests/jpa-mssql/**/*
+ - integration-tests/jpa-mysql/**/*
+ - integration-tests/jpa-postgresql/**/*
+ - integration-tests/jpa-without-entity/**/*
+ - integration-tests/common-jpa-entities/**/*
```
Comment (gsmet) [inline .github/labeler.yml]: As mentioned above, I think all of these should have the Hibernate ORM label.

File: .github/labeler.yml [hunk 6]
```diff
@@ -0,0 +1,273 @@
...
+
+area/qute:
+ - extensions/qute/**/*
```
Comment (gsmet) [inline .github/labeler.yml]: shouldn't you add the independent-project too?

File: .github/labeler.yml [hunk 7]
```diff
@@ -0,0 +1,273 @@
...
+area/reactive:
+ - extensions/reactive-mysql-client/**/*
+ - extensions/reactive-pg-client/**/*
+ - extensions/reactive-streams-operators/**/*
+ - integration-tests/reactive-mysql-client/**/*
+ - integration-tests/reactive-pg-client/**/*
+
+area/rest-client:
+ - extensions/rest-client/**/*
+ - extensions/elasticsearch-rest-client/**/*
```
Comment (gsmet) [inline .github/labeler.yml]: This one shouldn't be here but with the hibernate-search label. At least for now. It's not exposed, just an internal library of the Search extension.

File: .github/labeler.yml [hunk 8]
```diff
@@ -0,0 +1,273 @@
...
+
+area/qute:
+ - extensions/qute/**/*
+
+area/reactive:
+ - extensions/reactive-mysql-client/**/*
+ - extensions/reactive-pg-client/**/*
+ - extensions/reactive-streams-operators/**/*
+ - integration-tests/reactive-mysql-client/**/*
+ - integration-tests/reactive-pg-client/**/*
```
Comment (gsmet) [inline .github/labeler.yml]: Shouldn't also we have there all the `smallrye-reactive-messaging` ones?

File: .github/labeler.yml [hunk 9]
```diff
@@ -0,0 +1,273 @@
...
+ - integration-tests/vault/**/*
+ - integration-tests/vault-agroal/**/*
+ - integration-tests/vault-app/**/*
+
+area/vertx:
+ - extensions/vertx/**/*
+ - extensions/vertx-core/**/*
+ - extensions/vertx-graphql/**/*
+ - extensions/vertx-http/**/*
+ - extensions/vertx-keycloak/**/*
```
Comment (gsmet) [inline .github/labeler.yml]: This one should probably have the `security` label too.

File: .github/labeler.yml [hunk 10]
```diff
@@ -0,0 +1,273 @@
...
+ - extensions/vertx-web/**/*
+ - integration-tests/vertx/**/*
+ - integration-tests/vertx-graphql/**/*
+ - integration-tests/vertx-http/**/*
```
Comment (gsmet) [inline .github/labeler.yml]: Wondering if all these should have the `reactive` one?

File: .github/labeler.yml [hunk 11]
```diff
@@ -0,0 +1,273 @@
...
+### area/dependencies
+### area/dev-mode
+### area/gradle
+### area/housekeeping
+### area/infra
+### area/new-http
+### area/oidc-interoperability
+### area/openapi
+### area/platform
+### area/reactive-messaging
```
Comment (gsmet) [inline .github/labeler.yml]: This one should be used for the `smallrye-reactive-messaging*` extensions.

File: .github/labeler.yml [hunk 12]
```diff
@@ -0,0 +1,273 @@
...
+### area/security-consolidation
+### area/testing
```
Comment (gsmet) [inline .github/labeler.yml]: Maybe we can put this one on all the `test-framework` directory?

**Review by gsmet — CHANGES_REQUESTED** [review]
That's amazing, Max, thanks a lot for that one.

I added some comments inline.

File: .github/labeler.yml (hunk 2 above)
Comment (maxandersen) [reply .github/labeler.yml]: i'll probably write a new script to generate this stuff automatically ;) 
so can we say any folder with `jpa` in it should get the hibernate label ?

File: .github/labeler.yml (hunk 3 above)
Comment (maxandersen) [reply .github/labeler.yml]: yeah - after submitting it I realized the whole `devtools` system was missing which is actually my personal interest to get auto-labelled :) I'll get that added.

File: .github/labeler.yml (hunk 4 above)
Comment (maxandersen) [reply .github/labeler.yml]: where is that actually ? only spotting mongodb-client ....

File: .github/labeler.yml (hunk 5 above)
Comment (maxandersen) [reply .github/labeler.yml]: yup - I'll regen where "jpa" => hibernate

File: .github/labeler.yml [hunk 13]
```diff
@@ -0,0 +1,273 @@
...
+ - extensions/arc/**/*
+
+area/artemis:
+ - extensions/artemis-core/**/*
+ - extensions/artemis-jms/**/*
+ - integration-tests/artemis-core/**/*
+ - integration-tests/artemis-jms/**/*
+
+area/config:
+ - extensions/config-yaml/**/*
```
Comment (dmlloyd) [inline .github/labeler.yml]: I'd add `core/deployment/src/main/java/io/quarkus/deployment/configuration/**/*` and `core/runtime/src/main/java/io/quarkus/runtime/configuration/**/*` here as well.

**Comment by maxandersen** [comment]
actually - this is easier to update by hand.. update incoming soon(tm) :)

File: .github/labeler.yml (hunk 11 above)
Comment (maxandersen) [reply .github/labeler.yml]: done.

File: .github/labeler.yml (hunk 12 above)
Comment (maxandersen) [reply .github/labeler.yml]: done

File: .github/labeler.yml (hunk 13 above)
Comment (maxandersen) [reply .github/labeler.yml]: done

File: .github/labeler.yml (hunk 4 above)
Comment (maxandersen) [reply .github/labeler.yml]: ah - had to search a level deeper - extensions/panache/mongodb-*

File: .github/labeler.yml (hunk 8 above)
Comment (maxandersen) [reply .github/labeler.yml]: in addition to area/reactive-messaging ?

**Comment by maxandersen** [comment]
I've pushed an update and I wrote  a [jbang script](https://gist.github.com/maxandersen/590b8a0e824faeb3ee7ddfad741ce842) to verify this file. 

do the following in root:

```
$ sdk install jbang 
$ jbang https://git.io/JvJR5 .github/labeler.yml <optional path>
```

it has a few options you can see by running: `jbang https://git.io/JvJR5`

**Comment by gastaldi** [comment]
Make sure to add these files to be ignored by the azure pipeline

**Comment by maxandersen** [comment]
> Make sure to add these files to be ignored by the azure pipeline

done.

File: .github/labeler.yml [hunk 14]
```diff
@@ -0,0 +1,273 @@
...
+ - extensions/elytron-security-common/**/*
+ - extensions/elytron-security-jdbc/**/*
+ - extensions/elytron-security-oauth2/**/*
+ - extensions/elytron-security-properties-file/**/*
+ - integration-tests/elytron-security/**/*
+ - integration-tests/elytron-security-jdbc/**/*
+ - integration-tests/elytron-security-oauth2/**/*
+
+area/smallrye:
+ - extensions/smallrye-context-propagation/**/*
```
Comment (michalszynkiewicz) [inline .github/labeler.yml]: To me the specs are different worlds. I think JWT fits better in security, openapi could live together with resteasy, reactive-messaging-* in reactive

**Comment by gastaldi** [comment]
> > Make sure to add these files to be ignored by the azure pipeline
> 
> done.

@maxandersen I think you need to add your new files to https://github.com/quarkusio/quarkus/blob/master/azure-pipelines.yml#L13 to ignore it?

**Comment by maxandersen** [comment]
> @maxandersen I think you need to add your new files to https://github.com/quarkusio/quarkus/blob/master/azure-pipelines.yml#L13 to ignore it?

sorry was missing a push. done now.

File: .github/labeler.yml (hunk 14 above)
Comment (maxandersen) [reply .github/labeler.yml]: I'm not fully following - can you make a propsed edit here? otherwise we can update it after I push this initial setup.

**Comment by geoand** [comment]
Can you please rebase the PR onto master to hopefully make CI happy?

Thanks

**Pull request merged**